processor/src/trace/virtual_bus/circuit/prover.rs

```diff
@@ -289,10 +335,10 @@ fn prove_final_circuit_layer<
 
     // create the composed multi-linear for the second sum-check protocol using the randomness
     // sampled during the first one
-    let gkr_composition =
-        gkr_merge_composition_from_composition_polys(&composition_polys, r_sum_check, rand_merge);
+    let gkr_composition = GkrCompositionMerge::new(r_sum_check, rand_merge, log_up_randomness);
 
     // include the partially evaluated at the first sum-check randomness EQ multi-linear
+    // TODO: Find a better way than to push the evaluation of `EqFunction` here.
     mls.push(merged_mls[4].clone());
```

Comment:
Probably a `GkrOracle` trait which has a method to evaluate the Eq function and multiply by it when evaluating at a query.